jenkins/continuous-integration.jenkins

```diff
@@ -0,0 +1,41 @@
+pipeline {
+  agent {
+    docker { image 'node:10-slim' }
+  }
+  environment {
+    CODECOV_TOKEN = '523afe7a-4620-466f-a26a-d914c18a7c0a'
+  }
+  stages {
+    stage('Build') {
+      steps {
+        sh 'yarn'
+      }
+    }
+    stage('Test') {
+      parallel {
+        stage('Lint') {
+          steps {
+            sh 'yarn js-lint'
+            sh 'yarn typescript-lint'
+          }
+        }
+        stage('Type Checks') {
+          steps {
+            sh 'yarn type-check'
```

Comment:
Is this a prerequisite for running `yarn test`? I thought jest typechecks as it loads the `.ts` files for execution.

If it is a prerequisite, what is the advantage of making it a separate step? I think that combining it with `yarn test` (as a `pretest` entry in `package.json`) would make it easier for devs to run the tests (less documentation)

Reply:
in my experience i've had to run the type checker separately and it has given me different feedback than the tests and tslint. I'm not sure if there is a way to incorporate it into one of the other commands entirely but i'm totally down with hiding it behind package.json config